mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -269,23 +286,22 @@ private void save(MailboxSession mailboxSession, MessageIdMapper messageIdMapper
     }
 
     private Predicate<MailboxMessage> messageBelongsToUser(MailboxSession mailboxSession, MailboxMapper mailboxMapper) {
-        return Predicates.compose(
-            mailboxBelongsToUser(mailboxSession, mailboxMapper),
+        return compose(mailboxBelongsToUser(mailboxSession, mailboxMapper),
             MailboxMessage::getMailboxId);
     }
 
     private void allowOnMailboxSession(List<MailboxId> mailboxIds, MailboxSession mailboxSession, MailboxMapper mailboxMapper) throws MailboxNotFoundException {
-        Optional<MailboxId> mailboxForbidden = FluentIterable.from(mailboxIds)
-            .firstMatch(isMailboxOfOtherUser(mailboxSession, mailboxMapper))
-            .or(Optional.<MailboxId>absent());
+        Optional<MailboxId> mailboxForbidden = mailboxIds.stream()
+            .filter(isMailboxOfOtherUser(mailboxSession, mailboxMapper))
+            .findFirst();
 
         if (mailboxForbidden.isPresent()) {
             throw new MailboxNotFoundException("Mailbox with Id " + mailboxForbidden.get() + " does not belong to session");
         }
     }
 
     private Predicate<MailboxId> isMailboxOfOtherUser(MailboxSession mailboxSession, MailboxMapper mailboxMapper) {
-        return Predicates.not(mailboxBelongsToUser(mailboxSession, mailboxMapper));
+        return not(mailboxBelongsToUser(mailboxSession, mailboxMapper));
```

Comment:
in this case you can stick using Predicates, as it would allow you to not define yourself the not() method?

Reply:
Predicates is guava and it accept just guava's Predicate. That's why we have to define our-self method